### PR TITLE
Version 1.0.0

### PR DESCRIPTION
**Pull request by dushimeemma**
Description

This PR introduces `Version 1.0.0`

Type of change

[ ] Bug fix (Non breaking change that fixes an existing issue)

[x] New Feature (Non breaking change which adds funcitonality)

[ ] Breaking change (New feature that removes or causes an existing feature not to work as expected)

[ ] This change requires a documentation update

How Has This Been Tested? [ ] Unit tests

[ ] Component Tests

[ ] Integration Tests

How Should This Be Manually tested:

Clone this repo git clone https://github.com/dushimeemma/fs-simba-cal/.

Checkout the branch `git checkout main`

Run yarn install

Run yarn dev

Checklist: 
[x] My code follows all the styles and conventions outlined for this project

[x] I have linted my code before submission

[x] I have made my corresponding changes to the documentation

[x] My changes generates no new warning

[x] I have added tests that prove my feature is working or my fix is effective

[x] New and existing tests pass locally with my changes

N/A

### CODE REVIEW

**Comment by 447011** [comment]
I tried to log in with these credentials, however I'm getting this error:
username: simba@cal.rw
password: Dushime@2020^1

image

Another strange thing I noticed is that I'm logged in by default when I'm opening the app in incognito mode:
image

File: prisma/schema.prisma [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  user          User @relation(fields: [id], references: [id])
+  name          String
+  duration      Int
+  events        Events[]
+}
+
+model Events {
+  id            Int @default(autoincrement()) @id
+  eventType     EventTypes @relation(fields: [id], references: [id])
+  attendies     Attendees[]
```
Comment (447011) [inline prisma/schema.prisma:32]: typo (attendees)

File: prisma/schema.prisma [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  username      String?
+  email         String?   @unique
+  password      String?
+  eventTypes    EventTypes[]
+  createdAt     DateTime  @default(now()) @map(name: "created_at")
+  updatedAt     DateTime  @updatedAt @map(name: "updated_at")
+}
+
+model EventTypes {
+  id            Int       @default(autoincrement()) @id
```
Comment (447011) [inline prisma/schema.prisma:22]: the schema.prisma file looks like it's not formatted correctly. can you add a prettier plugin for it?

**Comment by 447011** [comment]
Can you remove the default `hello.ts` file?
image

File: pages/[user-slug]/[event-type-slug].tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
+import React, { useState, useEffect } from 'react';
+import Head from 'next/head';
+import { useRouter } from 'next/router';
+import AccessTimeTwoToneIcon from '@mui/icons-material/AccessTimeTwoTone';
+import PublicTwoToneIcon from '@mui/icons-material/PublicTwoTone';
+import KeyboardArrowDownOutlinedIcon from '@mui/icons-material/KeyboardArrowDownOutlined';
+
+import DatePicker from '../../components/shared/DatePicker';
```
Comment (447011) [inline pages/[user-slug]/[event-type-slug].tsx:8]: can you create an alias for that in the tsconfig? that we can access it via `@components/shared/DatePicker`

**Comment by 447011** [comment]
I just checked the files and it seems like there's actually no functionality for booking a meeting, right? 🤔

If not:
This should be a Minimum Viable Product (which should be usable) and the booking functionality is the most important feature in a scheduling app. It feels like you spent too much time on the design and on "low-prio" things instead of focusing on making the core feature work.

**Comment by 447011** [comment]
Any reason why you decided against using an auth library like `next-auth` for the first version?

I'd say it's fine for now but in the future we'll implement a "sign in with google" functionality. And for that I guess we'll need some sort of library. Or what do you think?

File: helpers/jwt_functions.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import jwt from 'jsonwebtoken';
```
Comment (447011) [inline helpers/jwt_functions.ts:1]: question regarding folder structure. there's now a `lib` folder and a `helpers` folder. what's difference? do we need both or can they be combined?

File: pages/[user-slug]/[event-type-slug].tsx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+const BookEvent = () => {
+  const router = useRouter();
+  const [value, setValue] = useState<Date | null>(new Date());
+  const [isSelectedValue, setIsSelectedValue] = useState(false);
```
Comment (447011) [inline pages/[user-slug]/[event-type-slug].tsx:13]: small thing and personal preference. I'd probably call it `isValueSelected`

**Comment by dushimeemma** [comment]
> I tried to log in with these credentials, however I'm getting this error: username: [simba@cal.rw](mailto:simba@cal.rw) password: Dushime@2020^1
> 
> image
> 
> Another strange thing I noticed is that I'm logged in by default when I'm opening the app in incognito mode: image

I realize the issue is the prisma I am going to work on it @danieltonel

File: pages/[user-slug]/[event-type-slug].tsx (hunk 5 above)
Comment (dushimeemma) [reply pages/[user-slug]/[event-type-slug].tsx:13]: I will make sure that in the future I have proper variable renaiming

File: pages/[user-slug]/[event-type-slug].tsx [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+            <span className='my-3'>{value?.toDateString()}</span>
+            {time.length &&
+              time.map((slug, index) => (
+                <button
+                  className='px-3 py-2 border-2 border-black w-[80%] hover:bg-primary hover:text-white flex justify-center items-center cursor-pointer my-2'
+                  key={index}
+                  type='button'
+                  onClick={() => {
+                    localStorage.setItem('date', `${value}`);
+                    localStorage.setItem('time', slug);
```
Comment (447011) [inline pages/[user-slug]/[event-type-slug].tsx:63]: why do we need to set the values to localStorage here?

File: helpers/jwt_functions.ts (hunk 4 above)
Comment (dushimeemma) [reply helpers/jwt_functions.ts:1]: we can combine both I think, I will change it in the future

File: pages/bookings.tsx [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+      }`
+    );
+  });
+  useEffect(() => {
+    const checkToken = localStorage.getItem('token');
+    if (checkToken === null) {
+      router.push('/auth/login');
+    }
+    setIsAuth(true);
+  }, [isAuth]);
```
Comment (447011) [inline pages/bookings.tsx:37]: this needs to be done on every protected page, right?

File: pages/bookings.tsx [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+    setTime(`${newTime.split(' ')[0]}`);
+    setNewTime(
+      `${newTime.split(':')[0]}:${
+        Number(newTime.split(':')[1].split(' ')[0]) + 15
+      }`
+    );
+  });
+  useEffect(() => {
+    const checkToken = localStorage.getItem('token');
+    if (checkToken === null) {
```
Comment (447011) [inline pages/bookings.tsx:33]: This is just a first draft right?

**Pull request merged**